src/Unicode/Utf8EncodingForm.dfy

```diff
@@ -249,9 +250,7 @@ module Utf8EncodingForm refines UnicodeEncodingForm {
     var z := (secondByte & 0xF) as bv24;
     var y := (thirdByte & 0x3F) as bv24;
     var x := (fourthByte & 0x3F) as bv24;
-    var v := (u1 << 18) | (u2 << 16) | (z << 12) | (y << 6) | x as Unicode.ScalarValue;
     assert {:split_here} true;
-    assert EncodeScalarValueQuadrupleByte(v) == m;
-    v
+    (u1 << 18) | (u2 << 16) | (z << 12) | (y << 6) | x as Unicode.ScalarValue
```

Comment:
This change is a bit scary too: the performance improve if we remove the assert that restates the postcondition?

Reply:
Apparently so! I'm not particularly worried that anything unsound is going on, if that's what you're worried about. I think the explanation is that Z3 is instantiating gobs of quantifiers, and in one case it gets lucky and in the other it doesn't.